src/main/java/io/vertx/httpproxy/interceptors/HeadersInterceptor.java

```diff
@@ -12,29 +12,65 @@
 package io.vertx.httpproxy.interceptors;
 
 import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.codegen.annotations.Unstable;
 import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.MultiMap;
 import io.vertx.httpproxy.ProxyInterceptor;
+import io.vertx.httpproxy.interceptors.impl.HeadersFilterInterceptorImpl;
 import io.vertx.httpproxy.interceptors.impl.HeadersInterceptorImpl;
 
 import java.util.Set;
+import java.util.function.Consumer;
 
 import static io.vertx.codegen.annotations.GenIgnore.PERMITTED_TYPE;
 
 @VertxGen
+@Unstable
 public interface HeadersInterceptor {
 
+  /**
+   * Apply callbacks to change the request and response headers when the proxy receives them.
+   *
+   * @param changeRequestHeaders the operation that applied to the request headers
+   * @param changeResponseHeaders the operation that applied to the response headers
+   * @return the created interceptor
+   */
+  @GenIgnore(PERMITTED_TYPE)
+  static ProxyInterceptor changeHeaders(Consumer<MultiMap> changeRequestHeaders, Consumer<MultiMap> changeResponseHeaders) {
```

Comment:
In Vert.x API, we use the Vert.x `Handler` type instead of `Consumer`. In this case, I think it shouldn't be necessary to annotate the method with `GenIgnore`.